homeassistant/components/zha/light.py

```diff
@@ -398,6 +389,16 @@ def __init__(self, unique_id, zha_device: ZhaDeviceType, channels, **kwargs):
                 effect_list.append(light.EFFECT_COLORLOOP)
                 if self._color_channel.color_loop_active == 1:
                     self._effect = light.EFFECT_COLORLOOP
+        self._attr_supported_color_modes = filter_supported_color_modes(
+            self._attr_supported_color_modes
+        )
+        if len(self._attr_supported_color_modes) == 1:
+            self._color_mode = next(iter(self._attr_supported_color_modes))
+        else:  # Light supports color_temp + hs, determine which mode the light is in
+            if self._color_channel.color_mode == LightColorMode.COLOR_TEMP:
```

Comment:
color_mode doesn’t exist on the channel. https://github.com/home-assistant/core/blob/371dfd85c8b61a62c48526778b377c090e6c0982/homeassistant/components/zha/core/channels/lighting.py#L25

It will need to be added. The attribute will also need to be added to the ZCL_INIT_ATTRS dict with `False` to make sure it’s read during initialization and not read from cache.

then the comparisons need to be against the Zigpy enum: https://github.com/zigpy/zigpy/blob/b166e2e1f22c49bad201d7ecdf202f0fc9b7b087/zigpy/zcl/clusters/lighting.py#L14